src/System.Private.ServiceModel/tests/Benchmarks/PerfUsingCoreWCFService/CoreWCFPerfService/Program.cs

```diff
@@ -0,0 +1,126 @@
+﻿using CoreWCF;
+using CoreWCF.Configuration;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+
+namespace WCFCorePerfService
+{
+    public class Program
+    {        
+        static void Main(string[] args)
+        {
+            string filePath = Path.Combine(Environment.CurrentDirectory, "WCFCorePerfService.exe");
+            Console.WriteLine(filePath);
+            string command = $" advfirewall firewall add rule name=\"WCFCorePerfService\" dir=in protocol=any action=allow program=\"{filePath}\" enable=yes";
+            ExecuteCommand(command, Environment.CurrentDirectory, TimeSpan.FromSeconds(20));
```

Comment:
A better option is to directly modify the firewall. You can find some old code which [does this here](https://github.com/dotnet/wcf/blob/v1.1.9/src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/PortManager.cs). It's basically a helper class which uses the COM component for programming the firewall and includes automatic cleanup of process exit. I believe this will work now as there's COM support in recent versions of .NET.

Reply:
I have use COM to add rules in the firewall. but this only support in windows OS